Check transaction method return types and allow non-void returns (#381)
vitusortner/floor: README.md

### Transactions
Whenever you want to perform some operations in a transaction you have to add the `@transaction` annotation to the method.

> It's also required to add the `async` modifier. These methods can only return `Future<>`.

Reviewer: 🔍 We could rephrase the second sentence in order to make the necessity of returning `Future`s sound less like a limitation but more a constraint or feature.

```suggestion
It's also required to add the `async` modifier. These methods have to return a `Future`.
```
